src/components/PostsList/PostsList.tsx

```diff
@@ -1,53 +1,83 @@
-import React from 'react';
-
+import React, { useState, useEffect } from 'react';
 import './PostsList.scss';
+import { getAllPosts, getPosts } from '../../api/posts';
+import { Post } from '../../react-app-env';
 
 type Props = {
-  posts: Post[],
-  postId: number,
-  setPostId: (id: number) => void,
+  userSelectedId: string;
+  selectPost: (arg?: Post) => void;
+  post: Post | undefined;
```

Comment:
I would recommend using the syntax type Props = { post?: Post; } which uses an optional parameter ?. This syntax emphasizes the fact that post can be an optional parameter, and allows you to avoid method call errors on an undefined value. 
In contrast, the syntax type Props = { post: Post | undefined; } uses the required post property, but allows its value to be undefined.